ext/rbs_extension/compat.h

```diff
@@ -0,0 +1,10 @@
+#ifdef __clang__
```

Comment:
Including `ruby.h` caused compilation errors without C23 extensions. This macro is to suppress diagnostics in the header files.